### PR TITLE
Create a track for tsdb data

### PR DESCRIPTION
**Pull request by nik9000**
Adds a new rally track for tsdb data. The data comes from a k8s cluster
and we anonymize it.

### CODE REVIEW

File: tsdb/track.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  "indices": [
+    {
+      "name": "tsdb",
+      "body": "index.json"
+    }
+  ],
+  "corpora": [
+    {
+      "name": "tsdb",
+      "#base-url": "https://rally-tracks.elastic.co/tsdb",
```
Comment (nik9000) [inline tsdb/track.json:15]: I've commented this out because I'd like a review on the tools first and then I'll upload the anonymized file.

**Comment by dliappis** [comment]
@michaelbaamonde I assigned this to you, since you're also working on CI for raly-tracks.

**Comment by pquentin** [comment]
@michaelbaamonde I had started the review already, assigning myself so that we don't duplicate the effort. But please feel free to add your thoughts!

**Review by pquentin — COMMENTED** [review]
The tools look good, thanks! As you said, I'd like to see the data now to test the track itself.

Were you looking for comments on the openness/anonymization balance or for someone running the script? I've not done either of those things, please tell me if you would like me to.

File: tsdb/_tools/anonymize.py [hunk 2]
```diff
@@ -0,0 +1,560 @@
...
+  if "AttachVolume.Attach succeeded for volume" in message:
+    return "AttachVolume.Attach succeeded for volume <snip>"
+  if "Scaled down replica set" in message:
+    return "Scaled down replica set <snip>"
+  if "Stopping container" in message:
+    return "Stopping container <snip>"
+  if "Deleted pod" in message:
+    return "Deleted pod <snip>"
+  if "Back-off pulling image" in message:
+    return "Back-off pulling image <snip>"
```
[the remark this answers is not in the document]
Comment (nik9000) [reply tsdb/_tools/anonymize.py:158]: :+1:

File: tsdb/_tools/anonymize.py [hunk 3]
```diff
@@ -0,0 +1,560 @@
...
+  raise ValueError("unsupported kubernetes.labels.kubernetes_io/arch [{0}]".format(v))
+
+def k8s_labels_k8s_os(v):
+  if v == "linux":
+    return v
+  raise ValueError("unsupported kubernetes.labels.kubernetes_io/os [{0}]".format(v))
+
+K8S_POD_STATUS_PHASE_PASSTHROUGH = {"failed", "pending", "running", "succeeded"}
+def k8s_pod_status_phase(v):
+  if v in K8S_POD_STATUS_PHASE_PASSTHROUGH:
```
[the remark this answers is not in the document]
Comment (nik9000) [reply tsdb/_tools/anonymize.py:308]: :+1:

**Comment by nik9000** [comment]
@pquentin I've applied that changes you requested.

**Comment by nik9000** [comment]
> Were you looking for comments on the openness/anonymization balance or for someone running the script? I've not done either of those things, please tell me if you would like me to.

I've talked about the openness/anonymization balance with some security folks and have asked them for a review too. If you are interested in running the script you can - I'll link you to the non-anonymized data in another channel. I was mostly hoping for a review on style and to double check that the script does what I say it does. I figure more eyes are better.

**Comment by michaelbaamonde** [comment]
I haven't yet had a chance to look over this in detail, but one thing that we need to figure out before merging is how to license this track. I'll start a conversation offline.

**Review by pquentin — APPROVED** [review]
Thanks, the technical part looks good to me! Left two nits.

I tested this successfully, and  got 45k docs/s on my laptop. I guess the only parts left to sort out are infosec and legal.

**Comment by nik9000** [comment]
I've never compressed the file! I'll fix that up.

File: tsdb/README.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+This track allows to overwrite the following parameters using `--track-params`:
+
+* `bulk_size` (default: 10000)
+* `bulk_indexing_clients` (default: 8): Number of clients that issue bulk indexing requests.
+* `ingest_percentage` (default: 100): A number between 0 and 100 that defines how much of the document corpus should be ingested.
+* `number_of_replicas` (default: 0)
+* `number_of_shards` (default: 1)
+* `force_merge_max_num_segments` (default: unset): An integer specifying the max amount of segments the force-merge operation should use.
+* `index_mode` (default: standard): Whether to make a standard index (`standard`) or time series index (`time_series`)
```
Comment (nik9000) [inline tsdb/README.md:120]: This should default to `time_series`.

File: tsdb/README.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
+## TSDB Track
+
+This data is anonymized monitoring data from elastic-apps designed to test
+our TSDB project. TSDB needs us to be careful how we anonymize. Too much
```
[the remark this answers is not in the document]
Comment (nik9000) [reply tsdb/README.md:4]: :+!:

File: tsdb/track.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      "documents": [
+        {
+          "source-file": "documents.json",
+          "document-count": 122613113,
+          "#compressed-bytes": 4820107188,
```
[the remark this answers is not in the document]
Comment (nik9000) [reply tsdb/track.json]: :+1:

**Comment by nik9000** [comment]
@pquentin I've pushed some changes, mostly picking up what you asked me to change, but also making it run in `time_series` mode by default and getting it up to date with ES's master branch. For the next while I think this'll be a "master branch only" sort of hting.

**Review by pquentin — APPROVED** [review]
Thanks!

**Pull request merged**